Example does not compile when using embedded graphics 0.8.0

Hello,

I'm having compiler errors when using a code like the m5atom_embedded_graphics.rs example. Rust is still a little criptic for me, so I cannot tell exactly what the  error means. Here it is when I use embedded-graphics 0.8.0:
```
error[E0599]: the method `translated` exists for struct `LedPixelDrawTarget<Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>`, but its trait bounds were not satisfied
  --> examples/ws2812_embedded_graphics.rs:32:40
   |
32 |         let mut translated_draw = draw.translated(Point::zero());
   |                                        ^^^^^^^^^^ method cannot be called due to unsatisfied trait bounds
   |
  ::: /home/xxxx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ws2812-esp32-rmt-driver-0.5.0/src/lib_embedded_graphics.rs:51:1
   |
51 | pub struct LedPixelDrawTarget<CDraw, CDev, S>
   | ---------------------------------------------
   | |
   | doesn't satisfy `<_ as Iterator>::Item = Pixel<_>`
   | doesn't satisfy `_: DrawTargetExt`
   | doesn't satisfy `_: DrawTarget`
   | doesn't satisfy `_: Iterator`
   | doesn't satisfy `_: PixelIteratorExt<_>`
   |
   = note: the following trait bounds were not satisfied:
           `<LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>> as Iterator>::Item = Pixel<_>`
           which is required by `LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: embedded_graphics::iterator::PixelIteratorExt<_>`
           `LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: Iterator`
           which is required by `LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: embedded_graphics::iterator::PixelIteratorExt<_>`
           `<&LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>> as Iterator>::Item = Pixel<_>`
           which is required by `&LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: embedded_graphics::iterator::PixelIteratorExt<_>`
           `&LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: Iterator`
           which is required by `&LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: embedded_graphics::iterator::PixelIteratorExt<_>`
           `LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: DrawTarget`
           which is required by `LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: embedded_graphics::draw_target::DrawTargetExt`
           `<&mut LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>> as Iterator>::Item = Pixel<_>`
           which is required by `&mut LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: embedded_graphics::iterator::PixelIteratorExt<_>`
           `&mut LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: Iterator`
           which is required by `&mut LedPixelDrawTarget<embedded_graphics_core::pixelcolor::rgb_color::Rgb888, LedPixelColorImpl<3, 1, 0, 2, 255>, LedPixelMatrix<28, 8>>: embedded_graphics::iterator::PixelIteratorExt<_>`

For more information about this error, try `rustc --explain E0599`.
warning: `pixel-board-firmware` (example "ws2812_embedded_graphics") generated 1 warning
```

I have read that are some (breaking) API changes between embedded-graphics versions. I have tried a number of versions and 0.7.1 works for me.

Thanks for your work in this library.

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -12,10 +12,10 @@ jobs:
 
     steps:
     - name: Install Rust for Xtensa
-      uses: esp-rs/xtensa-toolchain@v1.2
+      uses: esp-rs/xtensa-toolchain@v1.5
       with:
         default: true
-        version: "1.60.0"
+        version: "1.70.0"
         ldproxy: true
     - uses: actions/checkout@v3
       with:

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,7 +17,7 @@ unstable = []
 
 [dependencies]
 smart-leds-trait = { version = "0.2", optional = true }
-embedded-graphics-core = { version = "0.3", optional = true }
+embedded-graphics-core = { version = "0.4", optional = true }
 once_cell = "1"
 thiserror = "1"
 
@@ -26,10 +26,10 @@ esp-idf-sys = { version = ">=0.32", features = ["binstart"] }
 
 [dev-dependencies]
 smart-leds = "0.3"
-embedded-graphics = "0.7"
+embedded-graphics = "0.8"
 
 [build-dependencies]
-embuild = "0.30"
+embuild = "0.31"
 anyhow = "1"
 
 [profile.release]

--- a/src/driver/color.rs
+++ b/src/driver/color.rs
@@ -75,10 +75,18 @@ impl<
     #[inline]
     fn new_with_rgbw(r: u8, g: u8, b: u8, w: u8) -> Self {
         let mut array = [0; N];
-        array.get_mut(R_ORDER).and_then(|v| Some(*v = r));
-        array.get_mut(G_ORDER).and_then(|v| Some(*v = g));
-        array.get_mut(B_ORDER).and_then(|v| Some(*v = b));
-        array.get_mut(W_ORDER).and_then(|v| Some(*v = w));
+        if let Some(v) = array.get_mut(R_ORDER) {
+            *v = r;
+        }
+        if let Some(v) = array.get_mut(G_ORDER) {
+            *v = g;
+        }
+        if let Some(v) = array.get_mut(B_ORDER) {
+            *v = b;
+        }
+        if let Some(v) = array.get_mut(W_ORDER) {
+            *v = w;
+        }
         Self(array)
     }
 

--- a/src/driver/esp32_rmt.rs
+++ b/src/driver/esp32_rmt.rs
@@ -21,12 +21,12 @@ impl Ws2812Esp32RmtItemEncoder {
         let mut clock_hz = 0u32;
         esp!(unsafe { rmt_get_counter_clock(channel, &mut clock_hz as *mut u32) })?;
         let clock_hz = clock_hz as u64;
-        let to0h_clk = ((WS2812_TO0H_NS as u64) * clock_hz.clone() / 1000_000_000) as u32;
-        let to0l_clk = ((WS2812_TO0L_NS as u64) * clock_hz.clone() / 1000_000_000) as u32;
-        let to1h_clk = ((WS2812_TO1H_NS as u64) * clock_hz.clone() / 1000_000_000) as u32;
-        let to1l_clk = ((WS2812_TO1L_NS as u64) * clock_hz.clone() / 1000_000_000) as u32;
-        let bit0 = to0h_clk | (1 << 15) | (to0l_clk << 16) | (0 << 31);
-        let bit1 = to1h_clk | (1 << 15) | (to1l_clk << 16) | (0 << 31);
+        let to0h_clk = ((WS2812_TO0H_NS as u64) * clock_hz / 1_000_000_000) as u32;
+        let to0l_clk = ((WS2812_TO0L_NS as u64) * clock_hz / 1_000_000_000) as u32;
+        let to1h_clk = ((WS2812_TO1H_NS as u64) * clock_hz / 1_000_000_000) as u32;
+        let to1l_clk = ((WS2812_TO1L_NS as u64) * clock_hz / 1_000_000_000) as u32;
+        let bit0 = to0h_clk | (1 << 15) | (to0l_clk << 16);
+        let bit1 = to1h_clk | (1 << 15) | (to1l_clk << 16);
         Ok(Self { bit0, bit1 })
     }
 
@@ -58,7 +58,7 @@ unsafe extern "C" fn ws2812_rmt_adapter(
         return;
     }
 
-    let src_len = min(src_size, wanted_num / 8) as usize;
+    let src_len = min(src_size, wanted_num / 8);
     let src_slice = std::slice::from_raw_parts(src as *const u8, src_len);
     let dest_slice = std::slice::from_raw_parts_mut(dest, src_slice.len() * 8);
 

